gnucash/gnome/gnc-plugin-page-report.cpp

```diff
@@ -1793,17 +1792,37 @@ gnc_plugin_page_report_export_cb (GSimpleAction *simple,
                                      "document object with export-string or "
                                      "export-error."));
         }
-        result = TRUE;
     }
     else
-        result = gnc_html_export_to_file (priv->html, filepath);
-
-    if (!result)
     {
-        const char *fmt = _("Could not open the file %s. "
-                            "The error is: %s");
-        gnc_error_dialog (parent, fmt, filepath ? filepath : "(null)",
-                          strerror (errno) ? strerror (errno) : "" );
+        SCM chart_report = scm_call_1 (scm_c_eval_string ("gnc:report-chart?"),
```

Comment:
IMHO it's not necessary to trigger re-rendering for charts only. What's the harm in always repeating the render? Maybe in the future we'd include other JS libraries eg "ai-library.js" etc.

Reply:
> What's the harm in always repeating the render?

I imagine mostly the time it can take on some type of reports with large datasets? It's always a little bit concerning to take away a feature that exists because we don't know which users it's going to impact (if any). I can almost hear the complaint 'I have this mega-big report that already takes one minute to be generated and now when I try to export it's so slow!!!!".

For future maintainability, what if instead of testing the value of chart? here, I create a more generic helper function in report-core.scm, something like gnc:report-rerender-on-html-export? that can then be called here instead of gnc:report-chart? Right now the only test in that helper function would still be chart?, but then later if other features are added it will be possibly more obvious that this function has to be amended with whatever other tests are needed?

Reply:
> That's not easy to answer. I have a pretty large book--58MB uncompressed--and it can take almost a minute to create some reports. What I don't know is what exactly is taking the time, and since Guile's call stack is invisible to the profiler I can't find out.

Which reports? How about some good old fashioned `(display "checkpoint-A\n")`?

Reply:
> Which reports? How about some good old fashioned (display "checkpoint-A\n")?

The Net Worth barchart used to pretty reliably take over a minute, but I just tested it in 5.3 and it took only a couple of seconds.

> How about some good old fashioned (display "checkpoint-A\n")?

IO is slow so inserting any IO statements will distort the timing.

Reply:
> > Which reports? How about some good old fashioned (display "checkpoint-A\n")?
> 
> The Net Worth barchart used to pretty reliably take over a minute, but I just tested it in 5.3 and it took only a couple of seconds.

b9382d2cf22ae1549f144bd64b1fe9b7bfcaa3f5 is the likely culprit. Ditto #420.